GGProject/tests/test_api_emp.py

```diff
@@ -5,149 +5,104 @@
-            response.raise_for_status()
-        except req.exceptions.HTTPError as err:
-            print(err)
-
-        data = {
-            "first_name":"Caroline",
-            "last_name":"Orndorff",
-            "emp_ID":300,
-            "emp_email":"orndorffc@varentech.com",
-            "manager_email":"parksw@varentech.com",
-            "keycode":54321,
-            "emp_permissions":'2',
-            "last_login":"2019-06-27",
-        }
-        
-        try:
-            response = req.post(url + "employees/", json=data)
-            response.raise_for_status()
-        except req.exceptions.HTTPError as err:
-            print(err)
+        response = req.post(url + "employees/", json=data)
+        response.raise_for_status()
+        return response.json()
 
 
-    # Retrieve and display all employees
+    # Retrieve all employees
     def get_all_employees(self):
-        
-        url = str(self.live_server_url) + "/api/"
-
-        try:
-            response = req.get(url + "employees/")
-            response.raise_for_status()
-        except req.exceptions.HTTPError as err:
-            print(err)
 
-        content = response.json()
+        url = str(self.live_server_url) + "/api/"
 
-        for employee in content:
-            for key in employee:
-                print(key + ":", employee[key])
-            print()
-                
-        return response
+        response = req.get(url + "employees/")
+        response.raise_for_status()
+        return response.json()
 
 
     # Retrieve single employee based on ID
     def get_single_employee(self, emp_ID):
 
         url = str(self.live_server_url) + "/api/"
 
-        try:
-            response = req.get(url + "employees/" + str(emp_ID) + "/")
-            response.raise_for_status()
-        except req.exceptions.HTTPError as err:
-            print(err)
-
-        employee = response.json()
+        response = req.get(url + "employees/" + str(emp_ID) + "/")
+        response.raise_for_status()
+        return response.json()
 
-        for key in employee:
-            print(key + ":", employee[key])
-        print()
-
-        return response
-            
-
-    # Change employee's ID
-    def update_employee_ID(self, old_emp_ID, new_emp_ID):
-
-        url = str(self.live_server_url) + "/api/"
-
-        data = {
-            "emp_ID":new_emp_ID,
-        }
-
-        try:
-            response = req.put(url + "employees/" + str(old_emp_ID) + "/", json=data)
-            response.raise_for_status()
-        except req.exceptions.HTTPError as err:
-            print(err)
-
-        return response
-        
 
     # Delete single employee
     def delete_employee(self, emp_ID):
 
         url = str(self.live_server_url) + "/api/"
 
-        try:
-            response = req.delete(url + "employees/" + str(emp_ID) + "/")
-            response.raise_for_status()
-        except req.exceptions.HTTPError as err:
-            print(err)
```

Comment:
We may not want to print the error any more but do we still want to wrap this in a `try` block?

Reply:
Wouldn't that prevent the test from returning an error if something went wrong?

Reply:
If you wanted a more specific error message you could use `raise` and use a specific, possibly more useful error message, but if you want the error to be returned then I guess not having the `try` block is right

Reply:
Luckily the raise_for_status function will display the specific error by default.